apps/create-vesting/src/utils/decodeCalldata.ts

```diff
@@ -0,0 +1,109 @@
+import { decodeFunctionData, type Address, type PublicClient } from "viem";
+import type { CreateVestingFormData } from "./schema";
+import {
+  cowShedAbi,
+  vestingEscrowFactoryAbi,
+  weirollAbi,
+} from "@bleu/utils/transactionFactory";
+import { scaleToSecondsMapping } from "./scaleToSecondsMapping";
+
+export const decodeCalldata = async (
+  string: `0x${string}`,
+  publicClient: PublicClient,
+  user: Address,
+  tokenDecimals: number,
+): Promise<CreateVestingFormData> => {
+  const decodedFunctionData = decodeFunctionData({
+    abi: cowShedAbi,
+    data: string,
+  });
+
+  const result = {} as CreateVestingFormData;
+
+  if (!decodedFunctionData?.args)
+    throw new Error("error decoding cowShed calldata");
+
+  let createVestingData: {
+    args: readonly unknown[] | undefined;
+    functionName: string;
+  };
+
+  let weirollData: {
+    args: readonly unknown[] | undefined;
+    functionName: string;
+  };
+
+  // First scenario: vestUserInput -> decode create vesting contract
```

Comment:
Good comments, but I would wrap each decoding in a different function to help the user to understand each case.